src/App/Pages/Vault/ScanPageViewModel.cs

```diff
@@ -9,13 +17,46 @@ public class ScanPageViewModel : BaseViewModel
     {
         private bool _showScanner = true;
         private string _totpAuthenticationKey;
+        private IPlatformUtilsService _platformUtilsService;
+        private IDeviceActionService _deviceActionService;
+        private ILogger _logger;
 
         public ScanPageViewModel()
         {
-            ToggleScanModeCommand = new Command(() => ShowScanner = !ShowScanner);
+            ToggleScanModeCommand = new AsyncCommand(ToggleScanMode, onException: HandleException);
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>("platformUtilsService");
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>("deviceActionService");
+            _logger = ServiceContainer.Resolve<ILogger>();
+            InitAsync().FireAndForget();
         }
 
-        public Command ToggleScanModeCommand { get; set; }
+        public async Task InitAsync()
+        {
+            try
+            {
+                await Device.InvokeOnMainThreadAsync(async () =>
+                {
+                    var hasCameraPermission = await PermissionManager.CheckAndRequestPermissionAsync(new Permissions.Camera());
+                    HasCameraPermission = hasCameraPermission == PermissionStatus.Granted;
+                    ShowScanner = hasCameraPermission == PermissionStatus.Granted;
+                });
+
+                if (!HasCameraPermission)
+                {
+                    return;
+                }
+                InitScannerCommand.Execute(null);
+            }
+            catch (System.Exception ex)
+            {
+                HandleException(ex);
```

Comment:
In case an exception happens, does it make sense to stay on the screen? can the user retry somehow (if not it should go back showing some message)?

Reply:
you can always enter the code manually by typing even though the scanner is not working.